src/plugins/discourse/mirror.test.js

```diff
@@ -568,4 +654,548 @@ describe("plugins/discourse/mirror", () => {
+            "authorUsername": "credbot",
+            "cooked": "<h1>Hello World</h1>",
+            "id": 3,
+            "indexWithinTopic": 1,
+            "replyToPostIndex": null,
+            "timestampMs": 2003,
+            "topicId": 3,
+          },
+          Object {
+            "authorUsername": "credbot",
+            "cooked": "<h1>Hello World</h1>",
+            "id": 4,
+            "indexWithinTopic": 1,
+            "replyToPostIndex": null,
+            "timestampMs": 2004,
+            "topicId": 4,
+          },
+          Object {
+            "authorUsername": "credbot",
+            "cooked": "<p>New content, without bump</p>",
+            "id": 5,
+            "indexWithinTopic": 1,
+            "replyToPostIndex": null,
+            "timestampMs": 2005,
+            "topicId": 5,
+          },
+        ]
+      `);
+    });
+
+    it("mirrors supports the uncategorized category in recheckTopicsInCategories option", async () => {
+      const {mirror, fetcher, reporter} = example({
+        recheckTopicsInCategories: [1],
+      });
+
+      // Adds a category description #4.
+      // And a topic 5 which is a member of the #4 category.
+      fetcher.addPost(1, null);
+      fetcher.addPost(2, null);
+
+      // See we're trying to update #4 and #5.
+      const topicsCall = jest.spyOn(fetcher, "topicWithPosts");
+      await mirror.update(reporter);
+      const earlyCalls = [...topicsCall.mock.calls];
+      topicsCall.mockClear();
+      await mirror.update(reporter);
+
+      // The initial load has strict order requirements.
+      expect(earlyCalls).toEqual([[1], [2]]);
+      // The update does not.
+      expect(topicsCall).toHaveBeenCalledTimes(2);
+      expect(topicsCall).toHaveBeenCalledWith(1);
+      expect(topicsCall).toHaveBeenCalledWith(2);
+    });
+
+    it("doesn't recheck a topic more than once", async () => {
+      const {mirror, fetcher, reporter} = example({
+        recheckTopicsInCategories: [4],
+      });
+
+      // Adds a category description #4.
+      // And a topic 5 which is a member of the #4 category.
+      fetcher.addPost(1, null);
+      fetcher.addPost(2, null);
+      fetcher.addPost(3, null);
+      fetcher.addCategory(4);
+      fetcher.addPost(5, null);
+      fetcher.setTopicCategory(5, 4);
+      await mirror.update(reporter);
+
+      const topicsCall = jest.spyOn(fetcher, "topicWithPosts");
+      fetcher.addPost(1, 1);
+      fetcher.addPost(5, 1);
+      await mirror.update(reporter);
+
+      // #1 will be updated because of the new post bump.
+      // #4 will be updated because of the recheck of category.
+      // #5 will be updated because of the new post bump and recheck of category.
+      expect(topicsCall).toHaveBeenCalledTimes(3);
+      expect(topicsCall).toHaveBeenCalledWith(1);
+      expect(topicsCall).toHaveBeenCalledWith(4);
+      expect(topicsCall).toHaveBeenCalledWith(5);
+    });
+
+    it("only fetches new topics on `update`", async () => {
```

Comment:
You say it only fetches new topics... but if we had added a post to topic 1, the topic would not be "new", but it would get fetched, I think? So should the test description be "only fetches new or bumped topics"?

Reply:
You're right, this is a copy-pasted test from mode 1 and could be misinterpreted then as well, as new posts should be included too, which would be a bumped topic.

Reflecting what this test does, it("should not fetch other topics again when only a new topic was added between updates").

It isn't to say that new topics are the only type of topic the update function should look for. But that the scenario here is adding a new topic between updates and we're expecting update to detect it can leverage local data to only need to fetch that new topic.

Reply:
I've renamed this test (and in mode 1 in #1473)